src/Illuminate/Bus/DatabaseBatchRepository.php

```diff
@@ -283,10 +284,40 @@ protected function toBatch($batch)
             (int) $batch->pending_jobs,
             (int) $batch->failed_jobs,
             json_decode($batch->failed_job_ids, true),
-            unserialize($batch->options),
+            $this->unserialize($batch->options),
             CarbonImmutable::createFromTimestamp($batch->created_at),
             $batch->cancelled_at ? CarbonImmutable::createFromTimestamp($batch->cancelled_at) : $batch->cancelled_at,
             $batch->finished_at ? CarbonImmutable::createFromTimestamp($batch->finished_at) : $batch->finished_at
         );
     }
+
+    /**
+     * Serialize the given value.
+     *
+     * @param  mixed  $value
+     * @return string
+     */
+    protected function serialize($value)
+    {
+        $serialized = serialize($value);
+
+        return $this->connection instanceof PostgresConnection
+            ? base64_encode($serialized)
+            : $serialized;
+    }
+
+    /**
+     * Unserialize the given value.
+     *
+     * @param  string  $serialized
+     * @return mixed
+     */
+    protected function unserialize($serialized)
+    {
+        if ($this->connection instanceof PostgresConnection) {
+            $serialized = base64_decode($serialized, true) ?: $serialized;
```

Comment:
serialized string always minimum 2 chars `N;`. Second char always `:` or `;`. You can check second char for this.
```php
if ($this->connection instanceof PostgresConnection &&  ! in_array($serialized[1], [':', ';'], true)) {
```
Or copy as is from https://github.com/laravel/framework/blob/8.x/src/Illuminate/Cache/DatabaseStore.php#L387-L394

Reply:
right, I thought it's a little hacky therefore I used `$serialized = base64_decode($serialized, true) ?: $serialized`!